blog/2024-03-06-rag-tool-vercel-gen-ai/index.mdx

```diff
@@ -0,0 +1,443 @@
+---
+slug: rag-tool-vercel-gen-ai
+title: Building a RAG "tool" with Dewy and Vercel's Generative UI components
+description: A tutorial showing how to implement retrieval as an LLM-invoked tool with Vercel's Generative UI components library.
+image: screencap.gif
+keywords:
+  - RAG tool development
+  - Vercel Generative UI library
+  - Knowledge search application
+  - LLM-based knowledge retrieval
+  - Interactive UI elements
+  - Retrieval-Augmented Generation techniques
+  - Dewy knowledge base
+  - AI-driven application development
+  - Dynamic information retrieval
+  - Custom LLM tools implementation
+  - OpenAI API integration
+  - NextJS and React for AI applications
+  - Typescript AI projects
+  - Enhancing user experience with AI
+  - Generative UI design
+authors:
+  - ryan
+---
+# Building a RAG tool with Vercel's Generative UI components
```

Comment:
This seems to explain what RAG is and how to use it as a tool -- that has been well explored. I think it may be catchier to emphasis two or three tricky or surprising aspects of integrating with the new Vercel SDk -- for instance:

"The Vercel Generative UI SDK seems to focus on LLMs calling tools to display information to the user. At first glance, it may seem like this doesn't support Retrieval-Augmented Generation (RAG) tools. However, digging in a bit we can see that not only can it support the RAG tool, but the experience is actually really nice. While normal RAG chatbots are black boxes and only show the answer, using the Gen UI SDK it is easy and natural to show a widget containing the retrieved information as well as the generated answer."

etc.

Reply:
Probably some tone changes elsewhere, but it feels like something in this direction would be catchier -- not "here's another tutorial on RAG, this time with Vercel" but "here's a tutorial on how to do this thing in Vercel that seemed not possible, and how to make ti work".